lib/iris/_data_manager.py

```diff
@@ -0,0 +1,392 @@
+
+        Args:
+
+        * memo:
+            :class:`copy` memo dictionary.
+
+        Kwargs:
+
+        * data:
+            Replacement data to substitute the currently managed
+            data with.
+
+        * realised_dtype:
+            Replacement for the intended dtype of the realised lazy data.
+
+        Returns:
+            :class:`DataManager`
+
+        """
+        try:
+            if data is None:
+                # Copy the managed data.
+                if self.has_lazy_data():
+                    data = copy.deepcopy(self._lazy_array, memo)
+                else:
+                    data = self._real_array.copy()
+            else:
+                # Check that the replacement data is valid relative to
+                # the currently managed data.
+                DataManager(self.core_data).replace(data)
+
+            result = DataManager(data, realised_dtype=realised_dtype)
+        except ValueError as error:
+            emsg = 'Cannot copy {!r} - {}'
+            raise ValueError(emsg.format(type(self).__name__, error.message))
+
+        return result
+
+    def _dtype_setter(self, realised_dtype):
+        """
+        Set the intended dtype of the realised lazy data. This is to support
+        the case of lazy masked integral and boolean data in dask.
+
+        Args:
+
+        * realised_dtype:
+            A numpy :class:`~numpy.dtype`, array-protocol type string,
+            or built-in scalar type.
+
+        """
+        if realised_dtype is None:
+            self._realised_dtype = None
+        else:
+            realised_dtype = np.dtype(realised_dtype)
+            if self.has_real_data():
+                emsg = 'Cannot set realised dtype, no lazy data is available.'
+                raise ValueError(emsg)
+            if realised_dtype.kind not in 'biu':
+                emsg = ('Can only cast lazy data to an integer or boolean '
+                        'dtype, got {!r}.')
+                raise ValueError(emsg.format(realised_dtype))
+            self._realised_dtype = realised_dtype
+
+            # Check the manager contract, as the managed dtype has changed.
+            self._assert_axioms()
+
+    @property
+    def core_data(self):
+        """
+        If real data is being managed, then return the :class:`~numpy.ndarray`
+        or :class:`numpy.ma.core.MaskedArray`. Otherwise, return the lazy
+        :class:`~dask.array.core.Array`.
+
+        Returns:
+            The real or lazy data.
+
+        """
+        if self.has_real_data():
+            result = self._real_array
+        else:
+            result = self._lazy_array
+        return result
+
+    @property
+    def data(self):
+        """
+        Returns the real :class:`~numpy.ndarray` or
+        :class:`numpy.ma.core.MaskedArray`.
+
+        Returns:
+            The real data.
+
+        .. note::
+            Any lazy data being managed will be realised.
+
+        """
```

Comment:
I think this doc string need rewording. Usually the Returns section would have the data types.
Perhaps:
```
Returns the real data. If any lazy data is being managed, it will be realised.

Returns:
    the real :class:`~numpy.ndarray` or   :class:`numpy.ma.core.MaskedArray`

```